tests/tests_default.yml

```
@@ -2,6 +2,38 @@
 ---
 - name: Ensure that the role runs with default parameters
   hosts: all
-  gather_facts: false  # test that role works in this case
+  become: true
   roles:
-    - linux-system-roles.template
+    - linux-system-roles.journald
+
+  tasks:
+    - name: Get systemd-journald service state
+      command:
+        cmd: systemctl show --property SubState systemd-journald.service
+      register: systemctl
+      changed_when: false
+
+    - name: Verify that systemd-journald is running
+      assert:
+        that: systemctl.stdout == 'SubState=running'
+
+    - name: Verify that /run/log/journal/$MACHINE_ID/system.journal exists
+      command:
+        cmd: stat /run/log/journal/{{ ansible_machine_id }}/system.journal
+      changed_when: false
+
+    - name: Verify that system.journal is in online state
+      shell:
+        cmd: >
+          "journalctl --header
```

```suggestion
          set -euo pipefail
          journalctl --header
```
pipefail is required for shell that uses pipes
since you are using the `>` string operator, you do not need quotes (and in general you don't need to quote strings in yaml - https://blogs.perl.org/users/tinita/2018/03/strings-in-yaml---to-quote-or-not-to-quote.html)